src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java

```diff
@@ -0,0 +1,260 @@
+                if ( tranches.size() == 0 ) {
+                    throw new UserException(String.format("Couldn't find any tranche containing variants with a %s > %.2f. Are you sure the truth files contain unfiltered variants which overlap the input data?", metric.getName(), metric.getThreshold(trancheThreshold)));
+                }
+                break;
+            }
+
+            tranches.add(t);
+        }
+
+        tranches.sort(Tranche.TRUTH_SENSITIVITY_ORDER);
+        return tranches;
+    }
+
+    private static Tranche findTranche( final List<VariantDatum> data, final SelectionMetric metric, final double trancheThreshold, final VariantRecalibratorArgumentCollection.Mode model ) {
+        logger.info(String.format("  Tranche threshold %.2f => selection metric threshold %.3f", trancheThreshold, metric.getThreshold(trancheThreshold)));
+
+        double metricThreshold = metric.getThreshold(trancheThreshold);
+        int n = data.size();
+        for ( int i = 0; i < n; i++ ) {
+            if ( metric.getRunningMetric(i) >= metricThreshold ) {
+                // we've found the largest group of variants with sensitivity >= our target truth sensitivity
+                Tranche t = trancheOfVariants(data, i, trancheThreshold, model);
+                logger.info(String.format("  Found tranche for %.3f: %.3f threshold starting with variant %d; running score is %.3f ",
+                        trancheThreshold, metricThreshold, i, metric.getRunningMetric(i)));
+                logger.info(String.format("  Tranche is %s", t));
+                return t;
+            }
+        }
+
+        return null;
+    }
+
+    private static Tranche trancheOfVariants( final List<VariantDatum> data, int minI, double ts, final VariantRecalibratorArgumentCollection.Mode model ) {
+        int numKnown = 0, numNovel = 0, knownTi = 0, knownTv = 0, novelTi = 0, novelTv = 0;
+
+        double minLod = data.get(minI).lod;
+        for ( final VariantDatum datum : data ) {
+            if ( datum.lod >= minLod ) {
+                if ( datum.isKnown ) {
+                    numKnown++;
+                    if( datum.isSNP ) {
+                        if ( datum.isTransition ) { knownTi++; } else { knownTv++; }
+                    }
+                } else {
+                    numNovel++;
+                    if( datum.isSNP ) {
+                        if ( datum.isTransition ) { novelTi++; } else { novelTv++; }
+                    }
+                }
+            }
+        }
+
+        double knownTiTv = knownTi / Math.max(1.0 * knownTv, 1.0);
+        double novelTiTv = novelTi / Math.max(1.0 * novelTv, 1.0);
+
+        int accessibleTruthSites = VariantDatum.countCallsAtTruth(data, Double.NEGATIVE_INFINITY);
+        int nCallsAtTruth = VariantDatum.countCallsAtTruth(data, minLod);
+
+        return new Tranche(ts, minLod, numKnown, knownTiTv, numNovel, novelTiTv, accessibleTruthSites, nCallsAtTruth, model);
+    }
```

Comment:
These classes suffer from an almost complete lack of javadoc -- do you want to address that in this branch?